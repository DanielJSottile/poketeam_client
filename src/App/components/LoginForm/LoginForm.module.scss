@import '../../../index.scss';

.signup-form {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  label {
    font-family: $h3-font;
    letter-spacing: 3px;
    text-shadow: $med-dark-grey;
    color: $super-light-cream;
  }
}

.login-button {
  display: block;
  width: 200px;
  text-align: center;
  font-size: 16px;
  font-family: 'Montserrat Alternates', sans-serif;
  padding: 5px;
  border-radius: 40px;
  border: 5px double #444444;
  margin-left: auto;
  margin-right: auto;
  margin-top: 1rem;
  margin-bottom: 0;
}
