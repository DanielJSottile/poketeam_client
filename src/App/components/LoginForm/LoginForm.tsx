import React, {
  useContext,
  useState,
  FormEvent,
  FunctionComponent,
} from 'react';
import { faSignInAlt } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import toast from 'react-hot-toast';
import GeneralContext from '../../contexts/GeneralContext';
import AuthApiService from '../../services/auth-api-service';
import TokenService from '../../services/token-service';
import Button from '../Button';
import Input from '../Input';
import styles from './LoginForm.module.scss';

type LoginFormProps = {
  /** onLoginSuccess Function */
  onLoginSuccess: () => void;
};

const LoginForm: FunctionComponent<LoginFormProps> = ({ onLoginSuccess }) => {
  const { getUserState } = useContext(GeneralContext);
  const [userName, setUsername] = useState('');
  const [password, setPassword] = useState('');

  const handleSubmitJwtAuth = (ev: FormEvent<HTMLFormElement>) => {
    ev.preventDefault();

    AuthApiService.postLogin({
      user_name: userName,
      password: password,
    })
      .then((res) => {
        const successToast = () =>
          toast.success(`Login Successful: Welcome!! ${userName}`);
        successToast();
        setUsername('');
        setPassword('');
        TokenService.saveAuthToken(res.authToken);
        getUserState();
        onLoginSuccess();
      })
      .catch((res) => {
        const errorToast = () => toast.error(`Login Failed: ${res.error}!!`);
        errorToast();
      });
  };

  return (
    <form className={styles['signup-form']} onSubmit={handleSubmitJwtAuth}>
      <Input
        inputHasError={false}
        htmlFor='user_name'
        label='Username'
        placeholder='Username'
        autoComplete='username'
        type='text'
        name='user_name'
        id='user_name'
        onChangeCallback={(e) => {
          setUsername(e.target.value);
        }}
        value={userName}
      />
      <Input
        inputHasError={false}
        htmlFor='password'
        label='Password'
        placeholder='Password'
        type='password'
        autoComplete='current-password'
        name='password'
        id='password'
        onChangeCallback={(e) => {
          setPassword(e.target.value);
        }}
        value={password}
      />
      <Button type='submit' buttonClass={styles['login-button']}>
        Log In <FontAwesomeIcon icon={faSignInAlt} />
      </Button>
    </form>
  );
};

export default LoginForm;
