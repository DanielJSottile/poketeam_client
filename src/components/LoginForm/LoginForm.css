.signup-form {
  margin: auto;
  margin-top: 30px;
  margin-bottom: 30px;
}

form.signup-form label {
  font-family: 'Anton', sans-serif;
  letter-spacing: 3px;
  text-shadow: #444444;
  color: rgb(255, 255, 238);
}