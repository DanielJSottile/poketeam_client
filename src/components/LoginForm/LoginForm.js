import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import TokenService from '../../services/token-service';
import UserContext from '../../contexts/UserContext';
import AuthApiService from '../../services/auth-api-service';

export default class LoginForm extends Component {

  static contextType = UserContext;

  static defaultProps = {
    onLoginSucess: () => {}
  };

  state = {error: null}

  handleSubmitJwtAuth = ev => {
    ev.preventDefault();
    this.setState({error: null});
    const { user_name, password } = ev.target

    AuthApiService.postLogin({
      user_name: user_name.value,
      password: password.value,
    })
    .then(res => {
      user_name.value = ''
      password.value = ''
      TokenService.saveAuthToken(res.authToken)
      this.context.getUserState()
      this.props.onLoginSuccess()
    })
    .catch(res => {
      this.setState({error: res.error})
    })
  }

  render() {
    const { error } = this.state;

    return (
      <form className="signup-form"
      onSubmit={this.handleSubmitJwtAuth}
      >
        <div role='alert'>{error && <p className='red'>{error}</p>}</div>
        <div>
          <label htmlFor="user_name">Username</label>
          <input placeholder="Username" autoComplete="username" type="text" name="user_name" id="user_name" />
        </div>
        <div>
          <label htmlFor="password">Password</label>
          <input type="password" autoComplete="current-password" name="password" id="password" />
        </div>
        <button type="submit">Log In <i className="fas fa-sign-in-alt"></i></button>
        <p>Not A Member? </p>
        <Link to='/register'>Register Here:</Link>
      </form>
    );
  };
};

