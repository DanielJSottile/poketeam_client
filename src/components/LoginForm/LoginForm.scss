@import '../../index.scss';

.signup-form {
  margin: auto;
  margin-top: 30px;
  margin-bottom: 30px;
  
  label {
    font-family: $h3-font;
    letter-spacing: 3px;
    text-shadow: $med-dark-grey;
    color: $super-light-cream;
  }
}
