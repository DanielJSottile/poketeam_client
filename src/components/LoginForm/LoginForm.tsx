import React, { useContext, useState } from 'react';
import TokenService from '../../services/token-service';
import GeneralContext from '../../contexts/GeneralContext';
import AuthApiService from '../../services/auth-api-service';
import './LoginForm.scss';

// Component

const LoginForm = (props: any) => {

  // Set Context

  const GenCon = useContext(GeneralContext);

  // Set State

  const [state, setState] = useState({ error: null });

  // JWT Auth Functionality

  const handleSubmitJwtAuth = (ev: any) => {
    ev.preventDefault();
    setState((oldVals) => ({ ...oldVals, error: null }));
    const { user_name, password } = ev.target;

    AuthApiService.postLogin({
      user_name: user_name.value,
      password: password.value,
    })
      .then((res: any) => {
        user_name.value = '';
        password.value = '';
        TokenService.saveAuthToken(res.authToken);
        GenCon.getUserState();
        props.onLoginSuccess();
      })
      .catch((res) => {
        setState((oldVals) => ({ ...oldVals, error: res.error }));
      });
  };

  // Final Render

  return (
    <form className="signup-form" onSubmit={handleSubmitJwtAuth}>
      <div role="alert">
        {state.error && (
          <p className="error-login shake-horizontal">{state.error}</p>
        )}
      </div>
      <div>
        <label htmlFor="user_name">Username</label>
        <input
          placeholder="Username"
          autoComplete="username"
          type="text"
          name="user_name"
          id="user_name"
        />
      </div>
      <div>
        <label htmlFor="password">Password</label>
        <input
          type="password"
          autoComplete="current-password"
          name="password"
          id="password"
        />
      </div>
      <button type="submit">
        Log In <i className="fas fa-sign-in-alt"></i>
      </button>
    </form>
  );
};

export default LoginForm;
