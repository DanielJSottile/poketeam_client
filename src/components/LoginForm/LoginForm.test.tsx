import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
//import { render } from '@testing-library/react';
import LoginForm from './LoginForm';
import { shallow, mount, render } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('<LoginForm/>', (): void => {
  it('renders without crashing', (): void => {
    const div: HTMLDivElement = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
        <LoginForm/>
      </BrowserRouter>
      , div);
      ReactDOM.unmountComponentAtNode(div);
  });
  it('snapshot', (): void => {
    const wrapper = shallow(<LoginForm />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});