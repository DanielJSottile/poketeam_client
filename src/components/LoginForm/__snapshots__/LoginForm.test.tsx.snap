// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm/> snapshot 1`] = `
<form
  className="signup-form"
  onSubmit={[Function]}
>
  <div
    role="alert"
  />
  <div>
    <label
      htmlFor="user_name"
    >
      Username
    </label>
    <input
      autoComplete="username"
      id="user_name"
      name="user_name"
      placeholder="Username"
      type="text"
    />
  </div>
  <div>
    <label
      htmlFor="password"
    >
      Password
    </label>
    <input
      autoComplete="current-password"
      id="password"
      name="password"
      type="password"
    />
  </div>
  <button
    type="submit"
  >
    Log In 
    <i
      className="fas fa-sign-in-alt"
    />
  </button>
</form>
`;
